src/core/types/base/graph.ts

```diff
@@ -0,0 +1,189 @@
+import { isNativeAsset } from "./asset";
+import { Pool } from "./pool";
+
+export interface Graph {
+	vertices: Map<string, Vertex>;
+}
+
+type Vertex = {
+	name: string;
+	edges: Array<Edge>;
+};
+
+type Edge = {
+	from: Vertex;
+	to: Vertex;
+	pools: Array<Pool>;
+};
+
+// **************************************** GRAPH **************************************** \\
+
+/**
+ *
+ */
+export function newGraph(pools: Array<Pool>): Graph {
+	const vertices = new Map();
+	const graph: Graph = { vertices: vertices };
+	for (const pool of pools) {
+		const vertexA: Vertex = getVertex(
+			graph,
+			isNativeAsset(pool.assets[0].info)
+				? pool.assets[0].info.native_token.denom
+				: pool.assets[0].info.token.contract_addr,
+		);
+		const vertexB: Vertex = getVertex(
+			graph,
+			isNativeAsset(pool.assets[1].info)
+				? pool.assets[1].info.native_token.denom
+				: pool.assets[1].info.token.contract_addr,
+		);
+		connectTo(vertexA, vertexB, pool);
+		connectTo(vertexB, vertexA, pool);
+	}
+	return graph;
+}
+/**
+ *
+ */
+function getVertex(graph: Graph, name: string): Vertex {
+	const vertex = graph.vertices.get(name);
+	if (vertex === undefined) {
+		const addedVertex = newVertex(name);
+		graph.vertices.set(name, addedVertex);
+		return addedVertex;
+	} else {
+		return vertex;
+	}
+}
+
+// func (graph *Graph) GetPaths(start string, depth int) (paths []*Path) {
```

Comment:
Is this needed in the future or can we remove this?

Reply:
Good catch. Will remove from this pr and commit in another one with more explanation